### PR TITLE
Added Sporadic NaN, a robustness-tester block that randomly injects NaNs into buffers

### PR DESCRIPTION
**Pull request by ncorgan**

### CODE REVIEW

File: testers/SporadicNaN.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        // Calculate if a NaN will be injected.
+        const bool insertNaN = (_randomProb(_gen) <= _probability);
+
+        const auto &buffer = inputPort->takeBuffer();
+        inputPort->consume(inputPort->elements());
+
+        if(insertNaN)
+        {
+            // Scatter NaNs around the buffer.
+            static constexpr size_t numNaNs = 8;
```
Comment (ncorgan) [inline testers/SporadicNaN.cpp]: TODO: parameterize this

File: testers/SporadicNaN.cpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            return;
+        }
+
+        auto inputPort = this->input(0);
+        auto outputPort = this->output(0);
+
+        // Calculate if a NaN will be injected.
+        const bool insertNaN = (_randomProb(_gen) <= _probability);
+
+        const auto &buffer = inputPort->takeBuffer();
```
Comment (guruofquality) [inline testers/SporadicNaN.cpp]: this block is changing an input buffer, not just forwarding it or a part of it. cant do that, its not safe

File: testers/SporadicNaN.cpp (hunk 1 above)
Comment (ncorgan) [reply testers/SporadicNaN.cpp]: Done.

File: testers/SporadicNaN.cpp (hunk 2 above)
Comment (ncorgan) [reply testers/SporadicNaN.cpp]: Done.

**Comment by guruofquality** [comment]
Its funny, I ran into some -infs just today

**Pull request merged**